code/modules/mob/living/carbon/xenomorph/damage_procs.dm

```diff
@@ -157,41 +157,48 @@
 
 
 /mob/living/carbon/xenomorph/proc/check_blood_splash(damage = 0, damtype = BRUTE, chancemod = 0, radius = 1)
+
 	if(!damage)
 		return FALSE
-	var/chance = 20 //base chance
-	if(damtype == BRUTE)
-		chance += 5
-	chance += chancemod + (damage * 0.33)
+
 	var/turf/T = loc
+
 	if(!T || !istype(T))
 		return
 
+	var/chance = 20 + chancemod + (damage * 0.33)
+	if(damtype == BRUTE)
+		chance += 5
+
 	if(radius > 1 || prob(chance))
 
 		var/obj/effect/decal/cleanable/blood/xeno/decal = locate(/obj/effect/decal/cleanable/blood/xeno) in T
 
-		if(!decal) //Let's not stack blood, it just makes lagggggs.
-			add_splatter_floor(T) //Drop some on the ground first.
+		if(!decal)
+			add_splatter_floor(T)
 		else
-			if(decal.random_icon_states && length(decal.random_icon_states) > 0) //If there's already one, just randomize it so it changes.
+			if(decal.random_icon_states && length(decal.random_icon_states) > 0)
 				decal.icon_state = pick(decal.random_icon_states)
 
-		var/splash_chance = 40 //Base chance of getting splashed. Decreases with # of victims.
-		var/distance = 0 //Distance, decreases splash chance.
-		var/i = 0 //Tally up our victims.
+		var/victim_count = 0 //Tally up our victims.
+		for(var/mob/living/carbon/human/victim in range(radius,src))
+			var/distance = get_dist(src,victim)
+			var/splash_chance = 80 + (distance * -15) - (victim_count * 5)
+			if(victim.loc == loc)
+				splash_chance += 30
 
-		for(var/mob/living/carbon/human/victim in range(radius,src)) //Loop through all nearby victims, including the tile.
-			distance = get_dist(src,victim)
+			if(splash_chance > 0 && prob(splash_chance))
+				var/chosen_zone = pick("l_leg","r_leg","l_arm","r_arm","r_hand","l_hand","r_foot","l_foot","chest","head","groin")
+				var/datum/limb/E = victim.get_limb(chosen_zone)
+				var/splash_damage = rand(10,25) - victim.getarmor(chosen_zone, "acid")*0.5
```

Comment:
This is a weird way to handle it.
`take_damage_limb` has the `blocked` variable. Use it like other damage procs do, armor being a % damage reduction.

Reply:
No, this is fine and balanced.

Reply:
It's a flat damage reduction so suits with very high acid resist cannot take indirect acid damage from spraying sources. Even if they took a small amount, the amount of pain they'd get would be really high and it would make melee spec a pain.

Reply:
This is out of the standard, behaves differently than any other armor check.
Do the change, please.

Reply:
See the balance discord. Everyone thinks that taking acid damage in the first place is a bad idea and advocate for the outright removal of it.

Reply:
a method suggest in the discord was done